source/operators/captions/webcaptions.py

```diff
@@ -0,0 +1,1025 @@
+            raise MasExecutionError(self.operator_object.return_output_object())
+        else:
+            if response["Status"] == "Success":
+                return self.operator_object.return_output_object()
+            else:
+                self.operator_object.update_workflow_status("Error")
+                self.operator_object.add_workflow_metadata(
+                    WebCaptionsError="Unable to store captions {} {e}".format(webcaptions_operator_name, e=response))
+                raise MasExecutionError(self.operator_object.return_output_object())
+
+        metadata = {
+            "OperatorName": webcaptions_operator_name,
+            "WorkflowId": self.workflow_id,
+            "LanguageCode": language_code
+        }
+
+        return metadata
+
+    def GetWebCaptionsCollection(self):
+        webcaptions_collection_name = "TranslateWebCaptions"
+        print(webcaptions_collection_name)
+
+        response = dataplane.retrieve_asset_metadata(self.asset_id, operator_name=webcaptions_collection_name)
+        print(response)
+        return response["results"]["CaptionsCollection"]
+
+    def GetTextOnlyTranscript(self, language_code):
+
+        webcaptions = self.GetWebCaptions(language_code)
+        transcript = self.WebCaptionsToTextTranscript(webcaptions)
+
+        return transcript
+
+        # Convert most recently saved WebCaptions to a text only transcript
+
+    def WebCaptionsToTextTranscript(self, webcaptions):
+
+        transcript = ""
+
+        for caption in webcaptions:
+            transcript = transcript + caption["caption"]
+
+
+        return transcript
+
+    def PutWebCaptionsCollection(self, operator, collection):
+
+        collection_dict = {}
+        collection_dict["CaptionsCollection"] = collection
+        response = dataplane.store_asset_metadata(self.asset_id, self.operator_object.name, self.workflow_id,
+                                collection_dict)
+
+        if "Status" not in response:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="Unable to store captions collection metadata {e}".format(e=response))
+            raise MasExecutionError(self.operator_object.return_output_object())
+        else:
+            if response["Status"] == "Success":
+                self.operator_object.update_workflow_status("Complete")
+                return self.operator_object.return_output_object()
+            else:
+                self.operator_object.update_workflow_status("Error")
+                self.operator_object.add_workflow_metadata(
+                    WebCaptionsError="Unable to store captions collection {e}".format(e=response))
+                raise MasExecutionError(self.operator_object.return_output_object())
+
+    def WebCaptionsToSRT(self, webcaptions):
+        srt = ''
+
+        index = 1
+
+        for caption in webcaptions:
+
+            srt += str(index) + '\n'
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are concatenating a string inside a loop. Concatenating immutable sequences will always results in a new object. This means that building up a sequence by repeated concatenation will have a quadratic runtime cost in the total sequence length. For efficiency and to get a linear runtime cost, consider appending values to a list and concatenating that list using [str.join()](https://docs.python.org/3/library/stdtypes.html#str.join) or [bytes.join()](https://docs.python.org/3/library/stdtypes.html#bytes.join) method.

[Learn more](https://docs.python.org/3/faq/programming.html#what-is-the-most-efficient-way-to-concatenate-many-strings-together)